DBAgenda.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+use Monolog\Logger;
+use Sabre\VObject;
+
+abstract class DBAgenda extends Agenda {
+    protected $pdo;
+    
+    public function __construct(string $url, string $username, string $password, object $api, array $agenda_args) {
+        setLogHandlers($this->log);
+        
+        $this->caldav_client = new CalDAVClient($url, $username, $password);
+        $this->api = $api;
+
+        $this->openDB($agenda_args);
+        
+        $this->pdo->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
```

Comment:
When we read the code of this class it is not intuitive why we can use `$this->pdo` here.  To make it obvious, what about having `openDB` returns the pdo instead of setting it (so L15 above would look like `$this->pdo = $this->openDB($agenda_args);`

Reply:
eb96a32